choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java

```diff
@@ -289,9 +294,15 @@ public ChoreographerStep getStepById(final long id) {
 		Assert.notNull(action, "Action is null.");
 		
 		try {
-			choreographerActionRepository.refresh(action);
+			final Optional<ChoreographerAction> optional = choreographerActionRepository.findById(action.getId());
```

Comment:
input parameter should be actionId (long)
optional is a very generic name => should be action